src/elm/Page/Onboarding.elm

```diff
@@ -339,42 +395,53 @@ viewProfileInfo _ model =
     ]
 
 
-viewConsents : Environment -> Model -> List (Html Msg)
-viewConsents _ model =
+viewConsents : Environment -> Shared -> Model -> List (Html Msg)
+viewConsents _ shared model =
     [ Section.view
         [ Section.viewPaddedItem
-            [ H.p [] [ H.text "Vi trenger komme i kontakt med deg som reisende for å optimalisere opplevelsen av den nye nettbutikken. Vi blir veldig glade om samtykker til dette!" ]
+            [ H.p [] [ H.text "For å forbedre nettbutikken og dele relevant informasjon, ber vi om samtykke til å kontakte deg per e-post. Samtykkene kan du endre når som helst!" ]
             , H.p [] [ H.a [ A.href "https://beta.atb.no/private-policy", A.target "_blank" ] [ H.text "Les vår personvernerklæring (åpner nytt vindu)" ] ]
             ]
-        , Section.viewLabelItem "Samtykker"
-            [ Checkbox.init "consent1"
-                |> Checkbox.setChecked model.consent1
-                |> Checkbox.setOnCheck (Just ToggleConsent1)
-                |> Checkbox.setTitle "Jeg ønsker å bli kontaktet når endringer skjer i nettbutikken"
-                |> Checkbox.view
-            , Checkbox.init "consent2"
-                |> Checkbox.setChecked model.consent2
-                |> Checkbox.setOnCheck (Just ToggleConsent2)
-                |> Checkbox.setTitle "Jeg ønsker å delta på brukertester"
-                |> Checkbox.view
-            ]
-        , if model.consent1 || model.consent2 then
-            sectionTextInput "consent-email"
-                model.email
-                InputEmail
-                "E-postadresse"
-                "Legg inn din e-postadresse"
+        , Section.viewLabelItem "Velg samtykker" (List.map (viewConsent model) shared.consents)
+        , if not (Set.isEmpty model.consents) then
+            Section.viewItem
+                [ TextInput.init "consent-email"
+                    |> TextInput.setTitle (Just "E-postadresse")
+                    |> TextInput.setPlaceholder "Hvor kan vi kontakte deg?"
+                    |> TextInput.setOnInput (Just InputEmail)
```

Comment:
Å bruke samme input-boks fører til problemer med frem/tilbake ved at du kan lagre en epost i profilen din, så gå til consents, den er forhåndsutfylt (som er bra), men om man da endrer denne eposten og går tilbake til profil så er epost noe annet enn det som er lagret på personalia. Tror det er best å ha eget input felt som initieres ved nextstep til å være tilsvarende som lagret epost dersom det er lagret epost.

Reply:
Har lagt til egen state for consent email nå. Testet å gå frem/tilbake og det ser ut til å fungere som intended.